generators/spring-service/index.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2013-2018 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const _ = require('lodash');
+const BaseGenerator = require('generator-jhipster/generators/generator-base');
+const constants = require('generator-jhipster/generators/generator-constants');
+const SERVER_MAIN_SRC_DIR = `${constants.MAIN_DIR}kotlin/`;
+const SERVER_TEST_SRC_DIR = constants.SERVER_TEST_SRC_DIR; // TODO: Yet to change this  `${constants.TEST_DIR}kotlin/`;
+
+let useBlueprint;
+
+module.exports = class extends BaseGenerator {
+    constructor(args, opts) {
+        super(args, Object.assign({ fromBlueprint: true }, opts));
+        this.argument('name', { type: String, required: true });
+        this.name = this.options.name;
+    }
+
+    initializing() {
+        this.log(`The service ${this.name} is being created.`);
+        this.baseName = this.config.get('baseName');
+        this.packageName = this.config.get('packageName');
+        this.packageFolder = this.config.get('packageFolder');
+        this.databaseType = this.config.get('databaseType');
+    }
+
+    // Public API method used by the getter and also by Blueprints
+    _initializing() {
+        return {
+            initializing() {
+                this.log(`The spring-service ${this.name} is being created.`);
+                const configuration = this.getAllJhipsterConfig(this, true);
+                const blueprintConfiguration = this.getJhipsterAppConfig('generator-jhipster-kotlin');
+                this.baseName = blueprintConfiguration.baseName || configuration.get('baseName');
+                this.packageName = blueprintConfiguration.packageName || configuration.get('packageName');
+                this.packageFolder = blueprintConfiguration.packageFolder || configuration.get('packageFolder');
+                this.databaseType = blueprintConfiguration.databaseType || configuration.get('databaseType');
+
+            }
+        };
+    }
+
+    get initializing() {
+        if (useBlueprint) return;
```

Comment:
This is wrong, you should call the `super....` here instead of copying over things from base

Reply:
can you check the main generator please? This needs to be updated in `spring-controller`

Reply:
Can you be more specific please? Do you mean that both of the corresponding `spring-controller` and (currently under investigatioj) `spring-service` references are not implemented the proper way and that we should actually mimic the way it's done in the `generator-jhipster` project?

Reply:
So, both `spring-controller` and `spring-service` should better inherit their `initializing` function(ality) from their parent (`generator-base` of `generator-jhipster`).
That said and given that the remote `generator-jhipster` HEAD might have progressed a bit, if I just sync my local one to it and replace my `return this._initializing();` to something similar to `return super._initializing();`, everything should be fine, right?

_I am not behind my dev pc at the moment_